x-pack/plugins/secops/public/containers/hosts/index.tsx

```diff
@@ -48,20 +51,20 @@ type HostsProps = OwnProps & HostsComponentReduxProps;
 const HostsComponentQuery = pure<HostsProps>(
   ({
     id = 'hostsQuery',
-    query,
     children,
     filterQuery,
     sourceId,
     startDate,
     endDate,
     limit,
     poll,
+    type,
   }) => (
     <Query<
       GetHostsTableQuery.Query | GetHostSummaryQuery.Query,
       GetHostsTableQuery.Variables | GetHostSummaryQuery.Variables
     >
-      query={query}
+      query={type === hostsModel.HostsType.page ? HostsTableQuery : HostSummaryQuery}
```

Comment:
I traced the `updateLimit` all the way to over here. I understand the conditional logic and the touch points for when we add new routes that have different queries in this PR. It looks like it will work for when we need different queries for the Hosts Component and that's the main usage. For other queries which don't need it such as Authentication and UncommonProcessTable are just going to use their same stock query.

Down the road if we see a lot of usage this will be good. If not, I wonder down the road we would just split and copy the rest of this out. Still it's good clean readable code, so no changes requested.